test/config/utility.cc

```diff
@@ -400,6 +400,11 @@ void ConfigHelper::initializeTls(bool ecdsa_cert,
       TestEnvironment::runfilesPath("test/config/integration/certs/cacert.pem"));
   validation_context->add_verify_certificate_hash(TEST_CLIENT_CERT_HASH);
 
+  // We'll negotiate up to TLSv1.3, but it really depends on what the client
+  // sets.
+  common_tls_context.mutable_tls_params()->set_tls_maximum_protocol_version(
+      envoy::api::v2::auth::TlsParameters::TLSv1_3);
```

Comment:
I'm not in love with this change, since we don't have TLS 1.3 enabled by default yet. Could we get away with setting this only when needed?